designer/client/src/components/modals/Testing/TestingForm.tsx

```diff
@@ -0,0 +1,232 @@
+
+    // Elements
+    const testWithParametersElementWhenSingleSource =
+        value.testType === TestType.withParameters && !thereAreMultipleSources ? <MarkdownForm content={view.markdownContent} /> : <></>;
+
+    const testWithParametersElementWhenMultipleSources =
+        value.testType === TestType.withParameters && thereAreMultipleSources ? (
+            <Box
+                sx={{ display: "flex", flexWrap: "wrap", justifyContent: "center", gap: 2, width: "100%" }}
+                style={{ marginBottom: "12px", marginTop: "12px" }}
+            >
+                <Typography component="span" variant={"subtitle1"} noWrap={false} align={"center"}>
+                    {`Test with form is supported only for scenario with single source. Your scenario has ${sourcesFound} sources.`}
+                </Typography>
+            </Box>
+        ) : (
+            <></>
+        );
+
+    const testWithGeneratedDataElement =
+        value.testType === TestType.withGeneratedData ? (
+            <div style={{ marginBottom: "16px", marginTop: "28px" }}>
+                <FormLabel required>{t("testingForm.withGeneratedData.numberOfSamples.label", "Specify number of samples")}</FormLabel>
+                <div className={nodeValue} style={{ marginTop: "4px" }}>
+                    <NodeInput
+                        value={generatedDataTestSampleSize}
+                        onChange={(event) => setState({ generatedDataTestSampleSize: event.target.value })}
+                        className={nodeInput}
+                        autoFocus
+                    />
+                </div>
+                <ValidationLabels fieldErrors={getValidationErrorsForField(generatedDataTestingErrors, "testData")} />
+            </div>
+        ) : (
+            <></>
+        );
+
+    const buttons = (
+        <Box sx={{ display: "flex", justifyContent: "flex-end", gap: 1, width: "auto" }}>
+            <Button sx={{ width: "15%" }} size="medium" variant="outlined" onClick={() => closeDialog()}>
+                {t("testingForm.cancelButton.label", "Cancel")}
+            </Button>
+            <Button sx={{ width: "15%" }} size="medium" variant="contained" onClick={confirmForm} type="submit" disabled={!formIsValid}>
+                {t("testingForm.testButton.label", "Test")}
+            </Button>
+        </Box>
+    );
+
+    return (
+        <div className={cx(css({ paddingTop: 10, paddingBottom: 20 }))}>
+            <AdhocTestingFormContext.Provider
+                value={{
+                    value: adhocTestingCurrentValue,
+                    setValue: setAdhocTestingCurrentValue,
+                    parameters,
+                    variableTypes,
+                    errors: adhocTestingErrors,
+                }}
+            >
+                <NodeTable>
+                    <FormLabel required>{t("addProcessForm.label.testMode", "Data used in sources")}</FormLabel>
```

Comment:
```suggestion
                    <FormLabel required>{t("addProcessForm.label.testMode", "Data used in scenario sources")}</FormLabel>
```

Reply:
Ad 1. Done, added info in tooltip, as much as is currently available - 2 cases 
-    `Scenario testing is not supported for scenario in current state` - for example for archived, when the test button is disabled because of backed config (but FE does not have information about detailed reason)
- `Scenario testing is not supported for currently configured sources` - when the scenario sources do not support testing (or there are currently no sources)

Ad 2. I will update docs after this PR is merged

Ad. this comment: suggestion is applied

Reply:
Ad1: added tooltips for not available test types, with basic information, that the scenario sources do not support the test type.